Integrate with the updated VAL api utils.

Qubad786 opened the pull request:
[EDUCATOR-2171](https://openedx.atlassian.net/browse/EDUCATOR-2171)

Qubad786 commented:
@mushtaqak @muhammad-ammar @irfanuddinahmad that's the platform side of refactoring.

Qubad786 commented:
jenkins run quality

Qubad786 commented:
jenkins run quality

Qubad786 commented:
This is ready for the review.

Qubad786 commented:
Sandbox: https://studio-transcript-api-utils.sandbox.edx.org/container/block-v1:Arbi+CS101+2018_LT+type@vertical+block@7dc18d349beb4811a8f053fe59a9fd85?action=new

mushtaqak commented on cms/djangoapps/contentstore/views/transcripts_ajax.py:
```diff
@@ -173,26 +173,23 @@ def download_transcripts(request):
-                youtube_id_1_0=item.youtube_id_1_0,
-                html5_sources=item.html5_sources,
-            )
+            transcript = get_video_transcript_content(edx_video_id=item.edx_video_id, language_code=u'en')
```
Now we don't need youtube id and html5 sources?

Qubad786 replied:
Exactly!

mushtaqak commented on common/lib/xmodule/xmodule/video_module/transcripts_utils.py:
```diff
@@ -827,11 +849,7 @@ def get_transcripts_info(self, is_bumper=False, include_val_transcripts=False):
-                youtube_id_1_0=self.youtube_id_1_0,
-                html5_sources=self.html5_sources
-            )
+            transcript_languages = get_available_transcript_languages(edx_video_id=self.edx_video_id)
```
@Qubad786 Do we need to take care of TODO below now ?

Qubad786 replied:
@mushtaqak no, because S3 is still on fallback.

mushtaqak commented on requirements/edx/base.txt:
```diff
@@ -61,7 +61,7 @@ edx-proctoring==1.3.7
 edx-search==1.1.0
 edx-submissions==2.0.12
-edxval==0.1.10
+#edxval==0.1.10
```
Make sure to uncomment before merging.

Qubad786 replied:
okay.

mushtaqak commented on requirements/edx/github.txt:
```diff
@@ -100,6 +100,7 @@ git+https://github.com/edx/xblock-lti-consumer.git@v1.1.6#egg=lti_consumer-xbloc
 # This is here because all of the other XBlocks are located here. However, it is published to PyPI and will be installed that way
 xblock-review==1.1.4
 git+https://github.com/edx/edx-analytics-data-api-client.git@0.13.0#egg=edx-analytics-data-api-client==0.13.0
+-e git+https://github.com/edx/edx-val.git@mrehan/refactor-val-api-utils#egg=edxval
```
Make sure to remove before merging.

Qubad786 replied:
okay.

mushtaqak commented:
👍

Qubad786 commented on cms/djangoapps/contentstore/tests/test_transcripts_utils.py:
```diff
@@ -647,9 +647,9 @@ def test_convert_srt_to_sjson(self):
+        expected = self.sjson_transcript
         actual = transcripts_utils.Transcript.convert(self.srt_transcript, 'srt', 'sjson')
-        self.assertDictEqual(actual, expected)
+        self.assertDictEqual(json.loads(actual), json.loads(expected))
```
This is needed because Transcript.convert returns dumped transcript data. (in `transcripts_utils.py`)

Qubad786 commented:
@muhammad-ammar I've rebased and updated sandbox: https://studio-transcript-api-utils.sandbox.edx.org/container/block-v1:Arbi+CS101+2018_LT+type@vertical+block@7dc18d349beb4811a8f053fe59a9fd85?action=new

Qubad786 commented on common/lib/xmodule/xmodule/video_module/transcripts_utils.py:
```diff
@@ -527,47 +527,73 @@ def get_video_ids_info(edx_video_id, youtube_id_1_0, html5_sources):
+    """
+    name_and_extension = os.path.splitext(file_name)
+    basename, input_format = name_and_extension[0], name_and_extension[1][1:]
+    filename = '{base_name}.{ext}'.format(base_name=basename.encode('utf8'), ext=output_format)
```
This is to encode unicode characters with utf8 (if we dont do this, ascii will be used by default). For example, I've picked a video's client ID: "EBA101x Wk01 L07 Challenging English Sounds æ and e", `æ` here is unicode character and cannot be encoded with ascii.

*Example snippets*

1.
```bash
>> '{}'.format(u'EBA101x Wk01 L07 Challenging English Sounds æ and e')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
```

2.
```bash
>>'{}'.format(u'EBA101x Wk01 L07 Challenging English Sounds æ and e'.encode('utf8'))
'EBA101x Wk01 L07 Challenging English Sounds \xc3\xa6 and e'
```
